### PR TITLE
Add unit tests to test component and architecture filtering

### PR DESCRIPTION
**Pull request by quba42**
[noissue]

Having recently reworked these filter functions several times in a row to accommodate all kinds of edge cases, I feel like they would really benefit from some simple unit tests!

### CODE REVIEW

File: pulp_deb/tests/unit/test_sync_helpers.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+        self.assertEqual(
+            _filter_split_architectures(release_file_architectures, "bogus", "stable"),
+            ["all"],
+        )
+
+        release_file_architectures = "arm64 armel armhf i386 mips64el mipsel amd64"
+
+        self.assertEqual(
+            _filter_split_architectures(release_file_architectures, "bogus", "stable"),
```
Comment (mdellweg) [inline pulp_deb/tests/unit/test_sync_helpers.py:94]: Is the `release` argument even doing something in this function?

Comment (quba42) [reply pulp_deb/tests/unit/test_sync_helpers.py:94]: It is only used by the log message. It may do some actual work once my flat repo format changes are added on top.

**Comment by quba42** [comment]
Added tests for component filtering.

Still missing: Test for the log messages.

**Review by mdellweg — APPROVED** [review]
